### ISSUE TITLE
Mode 1 and gene features

### ISSUE THREAD

**Issue by Lavieenrose123** [huangyh09/brie#39]
Thank you for developing Brie. You gave multiple examples on https://brie.readthedocs.io/. All of these examples were based on mode 2 (cell features or aggregation). However, when it comes to mode 1, a file on gene features is needed (-g gene-feature-file). I don't find any related examples or descriptions in the current document. Could you please tell me how to generate the gene-feature-file for mode 1? What is the format of that file?

**Comment by huangyh09** [comment]
Hi, this is inherited from brie1, for which we produced some preprocessed gene features:
* https://brie.readthedocs.io/en/latest/brie1.html#brie-isoform-estimate
* human: https://sourceforge.net/projects/brie-rna/files/annotation/human/gencode.v25/
* mouse: https://sourceforge.net/projects/brie-rna/files/annotation/mouse/gencode.vM12/

On the other hand, we found that the mean of a cell population usually gives a good alternative prior compared to sequenced generated prior.

Yuanhua

**Comment by Lavieenrose123** [comment]
Thank you for your reply. For mode 2, I think, the fundemantal hypothesis is that the cell population is homogenous.  However, for certain disease (especially cancer), such hypothesis may not be reasonable. I want to quantify PSI on such highly hetergenous data. Do you think mode 1 is a better choice?

**Comment by huangyh09** [comment]
Good point. Mode 1 may have better flexibility in this scenario. Alternative, you can consider using cell types as covariates if using mode 2, which is equivalent to having a cell type-specific prior for each splicing event.

**Comment by horsedayday** [comment]
Thank you for your comments. For mode 2 with cell-type annotation as covariates, I have to cluster cells into different cell types using gene expression profiles first.  However, PSI itself can also be used to define cell types. For heterogeneous data, I should further explore if a count-defined cell population has homogenous splicing patterns. 

Suppose I use mode 1 to get the PSI matrix, then how to detect differential alternative splicing between psi-(count-)defined cell types? According to the manual, it seems that the DAS function is embedded into mode 2.

**Comment by Lavieenrose123** [comment]
Thank you for your comments. For mode 2 with cell-type annotation as covariates, I have to cluster cells into different cell types using gene expression profiles first.  However, PSI itself can also be used to define cell types. For heterogeneous data, I should further explore if a count-defined cell population has homogenous splicing patterns. 

Suppose I use mode 1 to get the PSI matrix, then how to detect differential alternative splicing between psi-(count-)defined cell types? According to the manual, it seems that the DAS function is embedded into mode 2.

**Comment by huangyh09** [comment]
Thanks for the further info. Yes, the DAS is embedded into mode 2 to avoid pair-wise cell comparison. For the DAS, wi would suggest direct using mode 2, as it detects whether the mean PSI is different between two cell groups.

If you want to use the PSI to perform downstream analysis, e.g., cell clustering, using gene sequence features may help, as it will avoid information leaking by using cell clusters.

**Comment by Lavieenrose123** [comment]
Thank you for the information.
